aea/skills/base.py

```diff
@@ -912,3 +765,356 @@ def _print_warning_message_for_non_declared_skill_components(
+        ] = {}
+
+        # populate indexes
+        for _path, component_classes in component_classes_by_path.items():
+            for (component_classname, _component_class) in component_classes:
+                type_ = self._get_skill_component_type(_component_class)
+                class_index.setdefault(component_classname, {}).setdefault(
+                    type_, set()
+                ).add(_component_class)
+
+        for component_type, by_id in declared_component_classes.items():
+            for component_id, component_config in by_id.items():
+                path = component_config.file_path
+                class_name = component_config.class_name
+                if path is not None:
+                    classes_in_path = component_classes_by_path[path]
+                    component_class_or_none: Optional[Type[SkillComponent]] = next(
+                        (
+                            actual_class
+                            for actual_class_name, actual_class in classes_in_path
+                            if actual_class_name == class_name
+                        ),
+                        None,
+                    )
+                    enforce(
+                        component_class_or_none is not None,
+                        self._get_error_message_prefix()
+                        + f"Cannot find class '{class_name}' for component '{component_id}' of type '{self._type_to_str(component_type)}' of skill '{self.configuration.public_id}' in module {path}",
+                    )
+                    component_class = cast(
+                        Type[SkillComponent], component_class_or_none
+                    )
+                    actual_component_type = self._get_skill_component_type(
+                        component_class
+                    )
+                    enforce(
+                        actual_component_type == component_type,
+                        self._get_error_message_prefix()
+                        + f"Found class '{class_name}' for component '{component_id}' of type '{self._type_to_str(component_type)}' of skill '{self.configuration.public_id}' in module {path}, but the expected type was {self._type_to_str(component_type)}, found {self._type_to_str(actual_component_type)} ",
+                    )
+                    used_classes.add(component_class)
+                    result.append(
+                        _SkillComponentLoadingItem(
+                            component_id,
+                            component_config,
+                            component_class,
+                            component_type,
+                        )
+                    )
+                else:
+                    # process the configuration at the end of the loop
+                    not_resolved_configurations[
+                        (component_type, component_id)
+                    ] = component_config
```

Comment:
here we process skill component configurations that have the file_path field specified (!=None). The others will be processed in the loop below.